_rules/error-field-is-not-identified-using-color-only

```diff
@@ -0,0 +1,107 @@
+---
+id:
+name: Error field is not identified using color differences only
+rule_type: atomic
+description: |
+  This rule checks that error fields can be identified not only by color differences but through another visual means.
+accessibility_requirements: # Remove whatever is not applicable
+  wcag21:1.4.1: # Use of Color (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  -  DOM Tree
+  -  CSS Styling
+acknowledgments:
+  authors:
+    -  Giacomo Petri
+---
+
+## Applicability
+
+This rule applies to any error field that is [visible][].
+
+## Expectation (1)
+
+Each test target, together with its visual context, identify the presence of errors not only through color differences.
+
+## Assumptions
+
+If the presence of errors is conveyed through the use of colors that differ not only in their hue, but that also have a difference in lightness and the difference in relative luminance between the colors leads to a contrast ratio of 3:1 or greater, then this counts as an additional visual mean.
+
+*Note*: It is possible for an error field to be distinguishable from adjacent fields but still not have a meaningful error message. In that case, it would pass Success Criterion 1.4.1: Use of Color but still fail [Success Criterion 3.3.1: Error Identification](https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html) and, if applicable, [Scuccess Criterion 3.3.3: Error Suggestion](https://www.w3.org/WAI/WCAG21/Understanding/error-suggestion.html).
+
+## Accessibility Support
+
+_No accessibility support issues known._
+
+## Background
+
+- [Understanding Success Criterion 1.4.1: Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)
+- [F81 - Failure of Success Criterion 1.4.1 due to identifying required or error fields using color differences only](https://www.w3.org/WAI/WCAG21/Techniques/failures/F81.html)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The presence of errors is conveyed through text error message.
+
+```html
+<label for="first_name">First Name</label>
+<input type="text" id="first_name" autocomplete="given-name" style="border:1px solid #dd0000" aria-invalid="true" aria-describedby="first_name_error_msg">
+<p id="first_name_error_msg" class="error_msg" style="color:#dd0000">This field is empty. Enter your first name.</p>
```

Comment:
I also feel this example (actually all examples) would require a non-errored field so that we have something to compare to.
Maybe this form is super bad and says "This field is empty" on every field, just in red for errors and green for valid (that would fail both 1.4.1 and 3.3.1, of course).

Reply:
Passed 2-3 and failed 1 examples already contain both input with errors and input without errors.
Fixed example 1.